shared/data/src/main/java/com/ivy/data/repository/impl/CategoryRepositoryImpl.kt

```diff
@@ -20,25 +20,40 @@ class CategoryRepositoryImpl @Inject constructor(
     private val dispatchersProvider: DispatchersProvider,
     private val writeEventBus: DataWriteEventBus,
 ) : CategoryRepository {
+
+    private val categoriesMemo = mutableMapOf<CategoryId, Category>()
+
     override suspend fun findAll(deleted: Boolean): List<Category> {
-        return withContext(dispatchersProvider.io) {
-            categoryDao.findAll(deleted).mapNotNull {
-                with(mapper) { it.toDomain() }.getOrNull()
+        return if (categoriesMemo.isNotEmpty()) {
```

Comment:
This will result in a bug if `findById()` is called before `findAll()` see how the `TagRepository solved it by using a boolean flag

Reply:
It actually makes a lot of sense. Fixed it. Also, fixed CI/CD failings.